lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -899,6 +935,15 @@ LogicalResult AndOp::canonicalize(AndOp op, PatternRewriter &rewriter) {
   if (narrowOperationWidth(op, true, rewriter))
     return success();
 
+  // and(a[0], a[1], ..., a[n]) -> icmp eq(a, -1)
```

Comment:
and reduction (andr) would be better.  comparison is more expensive.

Reply:
We don't have andr operation and `icmp eq(a, -1)` is emitted as `andr` in ExportVerilog. I think this is fine.